src/Tokenizer/Token.php

```diff
@@ -20,7 +20,10 @@
  * Representation of single token.
  * As a token prototype you should understand a single element generated by token_get_all.
  *
+ * @immutable
```

Comment:
> Marking the class as immutable helps phpstan to understand the immutability.

can you elaborate on that?
where phpstan gets lost on immutability of Token instance? there is no mutator method in this class.
happy to see phpstan error that is present without this change and solved after this change.

Reply:
sure - if mutator is added (or mutation reflection/closure rebind) PHPStan will warn newly

Reply:
thank you.
in that case, i would rather suggest to have `@readonly` annotation, applied for all DTO classes, and (as separate PR), have Fixer that converts annotation to `readonly class` syntax.

introducing `@immutable` in one place is not systematic approach

Reply:
Do you know `@immutable` is stronger?

https://phpstan.org/r/874645f0-1413-4f4c-a5ec-ab69a5958127

> i would rather suggest to have `@readonly` annotation, applied for all DTO classes, and (as separate PR), have Fixer that converts annotation to `readonly class` syntax.

I still run almost all projects with PHP 7.4 support so coding Fixer is out of my capacity.

Reply:
I do not believe it is stronger:
https://phpstan.org/r/874645f0-1413-4f4c-a5ec-ab69a5958127

please notice my previous suggestion of using `@readonly` should be only mid-step and we should convert it to `readonly class`.
(perhaps PHPStan could also treat `@readonly` as alias for @immutable`)

Reply:
yes, see https://github.com/phpstan/phpstan/issues/11775 for explanation why this PR is perfect and `@readonly` above the class(es) is currently useless :)

Reply:
Angly on me?